app/code/core/Mage/Log/Model/Visitor.php

```diff
@@ -258,6 +258,7 @@ public function saveByRequest($observer)
         }
 
         try {
+            $this->initServerData();
```

Comment:
The table `log_url_info` is updated here: https://github.com/OpenMage/magento-lts/blob/4b0be45ecfd569e3cc5716ec5ec2490c7559613c/app/code/core/Mage/Log/Model/Resource/Visitor.php#L69-L83

Instead of full init, perhaps we can selectively set the affected data:

```php
            $this->setHttpHost($this->_httpHelper->getHttpHost(true));
            $this->setRequestUri($this->_httpHelper->getRequestUri(true));
            $this->setHttpReferer($this->_httpHelper->getHttpReferer(true));
```

But I am not sure if this is the best, what do you think?

Reply:
@kiatng let me try to do this here my dear...

Reply:
@kiatng it worked! But, I think that duplicating the initServerData routines would not be an interesting alternative. That's why I kept the initServerData call inside the saveByRequest() method.